### PR TITLE
Makes black optional during the import time

### LINKED ISSUE #2604: `hypothesis --version` does not work
When I use `pip install hypothesis` in a clear env and run `hypothesis --version`, this happens:

```python
» hypothesis --version
Traceback (most recent call last):
  File "/Users/sobolev/Documents/github/returns/.venv/bin/hypothesis", line 6, in <module>
    from hypothesis.extra.cli import main
  File "/Users/sobolev/Documents/github/returns/.venv/lib/python3.7/site-packages/hypothesis/extra/cli.py", line 35, in <module>
    from hypothesis.extra import ghostwriter
  File "/Users/sobolev/Documents/github/returns/.venv/lib/python3.7/site-packages/hypothesis/extra/ghostwriter.py", line 57, in <module>
    import black
ModuleNotFoundError: No module named 'black'
```

Looks like `black` is treated as a required dependency right now.

After installing `black`:

```
» hypothesis --version
hypothesis, version 5.33.0
```

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #2604

### CODE REVIEW

**Comment by Zac-HD** [comment]
I would prefer to check that black is importable in the cli module, as we already do for click.

Thjs approach does allow the --version command to work, but otherwise delays the error and makes it somewhat more confusing.

Attempting to import black *in the write subcommand function* would have the advantages of both, I think.

**Comment by Zac-HD** [comment]
How about 

```diff
--- /dev/null
+++ hypothesis-python/RELEASE.rst
@@ -0,0 +1,7 @@
+RELEASE_TYPE: patch
+
+This patch improves the error message from the :command:`hypothesis write`
+command if :pypi:`black` (required for the :doc:`ghostwriter <ghostwriter>`)
+is not installed.
+
+Thanks to Nikita Sobolev for fixing :issue:`2604`!
diff --git a/hypothesis-python/src/hypothesis/extra/cli.py b/hypothesis-python/src/hypothesis/extra/cli.py
index 2c7a90c..858e07c 100644
--- hypothesis-python/src/hypothesis/extra/cli.py
+++ hypothesis-python/src/hypothesis/extra/cli.py
@@ -32,7 +32,14 @@ import importlib
 import sys
 from difflib import get_close_matches

-from hypothesis.extra import ghostwriter
+MESSAGE = """
+The Hypothesis command-line interface requires the `{}` package,
+which you do not have installed.  Run:
+
+    python -m pip install --upgrade hypothesis[cli]
+
+and try again.
+"""

 try:
     import click
@@ -40,16 +47,7 @@ except ImportError:

     def main():
         """If `click` is not installed, tell the user to install it then exit."""
-        sys.stderr.write(
-            """
-The Hypothesis command-line interface requires the `click` package,
-which you do not have installed.  Run:
-
-    python -m pip install --upgrade hypothesis[cli]
-
-and try again.
-"""
-        )
+        sys.stderr.write(MESSAGE.format("click"))
         sys.exit(1)


@@ -142,4 +140,10 @@ else:
         elif writer == "equivalent" and len(func) == 1:
             writer = "fuzz"

+        try:
+            from hypothesis.extra import ghostwriter
+        except ImportError:
+            sys.stderr.write(MESSAGE.format("black"))
+            sys.exit(1)
+
         print(getattr(ghostwriter, writer)(*func, except_=except_ or (), style=style))
```

as a minimal change?

File: hypothesis-python/src/hypothesis/extra/cli.py [hunk 1]
```diff
@@ -32,24 +32,21 @@
...
     python -m pip install --upgrade hypothesis[cli]
 
 and try again.
 """
-        )
+
+try:
+    import click
+except ImportError:
+    def main():
```
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/extra/cli.py:47]:
```suggestion
except ImportError:

    def main():
```

My bad, I accidentally deleted a blank line that the formatter insists on here :confused:

**Review by Zac-HD — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks for you help! 👍

**Pull request merged**